overlord/overlord.go

```diff
@@ -122,6 +130,17 @@ func (o *Overlord) Run() {
 	})
 }
 
+func (o *Overlord) ensureAfter(d time.Duration) {
+	time.AfterFunc(d, func() {
```

Comment:
That means we may get several Ensure calls right next to each other and then at arbitrary times. Feels a bit too loose. It'd be nicer to see EnsureAfter as _"no later than after D"_. On the first Ensure call, they have a chance to schedule it again if they care. If we run Ensure and nobody schedules, we use the default interval next.

We can easily implement that in terms of a single timer:
```
type Overlord struct {
        ensureLock  sync.Mutex
        ensureNext  time.Time
        ensureTimer *time.Timer
}
```

ensureAfter becomes `if now+d < ensureNext { ensureTimer.Reset(d); ensureNext = now+d }`.

Right _before_ calling Ensure we simply `ensureTimer.Reset(defaultInterval)`. Needs to be before so that we respect calls with EnsureAfter on a short duration inside the Ensure handling itself.